R/download.R

```diff
@@ -0,0 +1,162 @@
+#' Retrieve value from key
+#'
+#' @description
+#' Retrieve value from key included in configuration file
+#'
+#' @param key character key
+#'
+#' @return character containing value
+#'
+#' @keywords internal
+retrieve_value_key <- function(key) {
+  config_file <- system.file(
+    "extdata",
+    "config.yaml",
+    package = "ColOpenData",
+    mustWork = TRUE
+  )
+  key_path <- config::get(
+    value = key,
+    file = config_file
+  )
+  if (rlang::is_empty(key_path)) {
+    stop("information is not available")
+  }
+  return(key_path)
+}
+
+#' Retrieve path of named dataset
+#'
+#' @param dataset character with the dataset name
+#'
+#' @description
+#' Datasets are included in the general documentation file. Only dictionaries
+#' and support files are treated differently and not included in the file.
+#'
+#' @return character with path to retrieve the dataset from server
+#'
+#' @keywords internal
+retrieve_path <- function(dataset) {
+  checkmate::assert_character(dataset)
+
+  # nolint start: nonportable_path_linter
+  base_path <- retrieve_value_key("base_path")
+  if (grepl("DICT", dataset)) {
+    group <- "dictionaries"
+    group_path <- retrieve_value_key(group)
+    ext <- ".csv"
+    dataset_path <- paste0(dataset, ext)
+    file_path <- file.path(base_path, group_path, dataset_path)
+  } else {
+    all_datasets <- list_datasets()
+    dataset_info <- all_datasets[which(all_datasets$name == dataset), ]
+    if (nrow(dataset_info) == 1) {
```

Comment:
We tried to only use one function, since it was possible and avoided code repetition, however, the readability is bad (as you have said, too many nested loops). We separated the functions, to make the whole process more readable and mantainable. This is included in 9c2f8de749c4d70a8b07635691cd1906fce05eeb